GDJS/Runtime/howler-sound-manager/howler-sound-manager.js

```diff
@@ -14,87 +14,264 @@
-    that._paused = true;
-    that._stopped = false;
-  });
-};
-gdjs.HowlerSound.prototype = Object.create(Howl.prototype);
+  /**
+   * Pauses playback of the sound, saving the seek of playback.
+   * @returns {HowlerSound} The current instance for chaining.
+   */
+  pause() {
+    if (this._id === null) return this;
+    this._paused = true;
+    this._stopped = false;
+    this._howl.pause(this._id);
+    return this;
+  }
 
-// Redefine `stop`/`play`/`pause` to ensure the status of the sound
-// is immediately updated (so that calling `stopped` just after
-// `play` will return false).
+  /**
+   * Stops playback of the sound, resetting seek to 0.
+   * @returns {HowlerSound} The current instance for chaining.
+   */
+  stop() {
+    if (this._id === null) return this;
+    this._paused = false;
+    this._stopped = true;
+    this._howl.stop(this._id);
+    return this;
+  }
 
-gdjs.HowlerSound.prototype.stop = function () {
-  this._paused = false;
-  this._stopped = true;
-  return Howl.prototype.stop.call(this);
-};
-gdjs.HowlerSound.prototype.play = function () {
-  this._paused = false;
-  this._stopped = false;
-  return Howl.prototype.play.call(this);
-};
-gdjs.HowlerSound.prototype.pause = function () {
-  this._paused = true;
-  this._stopped = false;
-  return Howl.prototype.pause.call(this);
-};
+  /**
+   * Check if the sound is currently playing.
+   * @returns {boolean}
+   */
+  playing() {
+    if (this._id === null) return false;
+    return this._howl.playing(this._id);
+  }
 
-// Add methods to query the status of the sound:
+  /**
+   * Check if the sound is currently paused.
+   * @returns {boolean}
+   */
+  paused() {
+    return this._paused;
+  }
 
-gdjs.HowlerSound.prototype.paused = function () {
-  return this._paused;
-};
-gdjs.HowlerSound.prototype.stopped = function () {
-  return this._stopped;
-};
-gdjs.HowlerSound.prototype.canBeDestroyed = function () {
-  return this._canBeDestroyed;
-};
+  /**
+   * Check if the sound is currently stopped.
+   * @returns {boolean}
+   */
+  stopped() {
+    return this._stopped;
+  }
+
+  /**
+   * Check if the sound can be deleted to free memory.
+   * This becomes true when the sound playback has ended.
+   * @returns {boolean}
+   */
+  canBeDestroyed() {
+    return this._canBeDestroyed;
+  }
 
-// Methods to safely update the rate of the sound:
+  /**
+   * Get the sound rate
+   * @returns {number} The current rate.
+   * @deprecated Use rate instead.
+   */
+  getRate() {
+    return this.rate();
+  }
 
-gdjs.HowlerSound.prototype.getRate = function () {
-  return this._rate;
-};
-gdjs.HowlerSound.prototype.setRate = function (rate) {
-  this._rate = gdjs.HowlerSoundManager.clampRate(rate);
-  this.rate(this._rate);
+  /**
+   * Set the sound rate
+   * @param {number} rate The new rate.
+   * @deprecated Use rate instead.
+   */
+  setRate(rate) {
+    this.rate(rate);
+  }
+  /**
+   * @returns {number}
+   */
+
+  /**
+   * Get/set the sound rate.
```

Comment:
I'm not a big fan of these mix of setter/getter in the same function. 
They are hard/impossible to type statically, and they are probably a nightmare to compile to machine code by JIT because their argument always differ... even their return value differ.
They also force you to do ugly `typeof rate === 'undefined'`.

Let's remove all of these and use getRate/setRate instead. Simple, efficient, can be statically verified, and setters can be chained. What's not to love? 😄

Reply:
> What's not to love?

I wanted to keep it consistent with howler. Should I also leave out chaining on setters?

Reply:
I guess the reason for having them is for keeping compatibility with Howler though, as you said... I wonder if it's worth doing it though.

Reply:
> Should I also leave out chaining on setters?

Chaining on setter is fine (we can type it, it's pleasant to use, there are no optimization problems).
I think we should go for "classic" getter and chainable setters. If we all document them, it will be an easy to use API for users. If some people were using them in JS code as `Howl`, we can always add compatibility function (a `rate` function that would use getRate/setRate, etc...) - but only if someone really complains about this (to avoid cluttering our sound API).